src/views/Project/index.jsx

```diff
@@ -1,111 +1,102 @@
 import { hot } from 'react-hot-loader';
 import { Component, Fragment } from 'react';
 import Typography from '@material-ui/core/Typography';
+import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
+import _ from 'lodash';
 import projects from '../../data/loader';
+import Spinner from '../../components/Spinner';
+import ErrorPanel from '../../components/ErrorPanel';
 import TasksTable from '../../components/TasksTable';
+import Issues from './issues.graphql';
 
-const data = [
-  {
-    project: 'Servo',
-    summary: '1436212 - Add pagination to listClients',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-05-20',
-  },
-  {
-    project: 'Taskcluster',
-    summary:
-      '1443017 - Use a mock AWS library to test publishing API definitions',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-05-14',
-  },
-  {
-    project: 'Servo',
-    summary: '1443016 - Create a fake version of azure-blob-storage',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-05-12',
-  },
-  {
-    project: 'Servo',
-    summary: '1441960 - Add measure of time to start processing a task',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-05-06',
-  },
-  {
-    project: 'Taskcluster',
-    summary: '1451548 - Return 404 for indexes and namespaces that are expired',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-05-04',
-  },
-  {
-    project: 'Taskcluster',
-    summary: '1441977 - Run tests for taskcluster-treeherder in Taskcluster',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-05-03',
-  },
-  {
-    project: 'Servo',
-    summary: '1446768 - Only post "No taskcluster jobs.." to a PR once',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-05-01',
-  },
-  {
-    project: 'Taskcluster',
-    summary:
-      '1457126 - Authorization failures should state which clientId lacks scopes',
-    tag: 'Rust',
-    assignee: 'None',
-    lastUpdated: '2018-04-05',
-  },
-  {
-    project: 'Taskcluster',
-    summary: '44 - Add pagination to auth.listRoles',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-03-05',
-  },
-  {
-    project: 'Taskcluster',
-    summary:
-      '1306494 - Add a diff+commit submit button to some text areas in tc-tools',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-02-12',
-  },
-  {
-    project: 'Taskcluster',
-    summary:
-      '333 - Make azure-entities and azure-blob0-storage independent of tc-lib-monitor',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-02-04',
-  },
-  {
-    project: 'Taskcluster',
-    summary: '1344912 - Support tag events, too',
-    tag: 'JS',
-    assignee: 'None',
-    lastUpdated: '2018-01-01',
-  },
-  {
-    project: 'Taskcluster',
-    summary: '1453714 - Return http 424 instead of 403 for error artifacts',
-    tag: 'Python',
-    assignee: 'None',
-    lastUpdated: '2017-12-04',
-  },
-];
+const githubQuery = fileName => {
```

Comment:
This function is too complex for me to understand what is happening just by looking at it. Could you break it up into several smaller functions?